File: bindings/lua/Polycode/MouseEventResult.lua
class "MouseEventResult"



function MouseEventResult:__getvar(name)
	if name == "hit" then
		return Polycore.MouseEventResult_get_hit(self.__ptr)
	elseif name == "blocked" then
		return Polycore.MouseEventResult_get_blocked(self.__ptr)
	end
end


function MouseEventResult:__setvar(name,value)
	if name == "hit" then
		Polycore.MouseEventResult_set_hit(self.__ptr, value)
		return true
	elseif name == "blocked" then
		Polycore.MouseEventResult_set_blocked(self.__ptr, value)
		return true
	end
	return false
end


function MouseEventResult:__delete()
	if self then Polycore.delete_MouseEventResult(self.__ptr) end
end

File: bindings/javascript/Polycode/MouseEventResult.js
function MouseEventResult() {
	Object.defineProperties(this, {
		'hit': { enumerable: true, configurable: true, get: MouseEventResult.prototype.__get_hit, set: MouseEventResult.prototype.__set_hit},
		'blocked': { enumerable: true, configurable: true, get: MouseEventResult.prototype.__get_blocked, set: MouseEventResult.prototype.__set_blocked}
	})
}
MouseEventResult.prototype.__get_hit = function() {
	return Polycode.MouseEventResult__get_hit(this.__ptr)
}

MouseEventResult.prototype.__set_hit = function(val) {
	Polycode.MouseEventResult__set_hit(this.__ptr, val)
}

MouseEventResult.prototype.__get_blocked = function() {
	return Polycode.MouseEventResult__get_blocked(this.__ptr)
}

MouseEventResult.prototype.__set_blocked = function(val) {
	Polycode.MouseEventResult__set_blocked(this.__ptr, val)
}

Duktape.fin(MouseEventResult.prototype, function (x) {
	if (x === MouseEventResult.prototype) {
		return;
	}
	Polycode.MouseEventResult__delete(x.__ptr)
})
